lighthouse-core/gather/computed/trace-of-tab.js

```diff
@@ -45,18 +45,30 @@ class TraceOfTab extends ComputedArtifact {
       .filter(e => {
         return e.cat.includes('blink.user_timing') ||
           e.cat.includes('loading') ||
-          e.cat.includes('devtools.timeline') ||
-          e.name === 'TracingStartedInPage';
+          e.cat.includes('devtools.timeline');
       })
       // @ts-ignore - stableSort added to Array by WebInspector.
       .stableSort((event0, event1) => event0.ts - event1.ts);
 
-    // The first TracingStartedInPage in the trace is definitely our renderer thread of interest
-    // Beware: the tracingStartedInPage event can appear slightly after a navigationStart
-    const startedInPageEvt = keyEvents.find(e => e.name === 'TracingStartedInPage');
-    if (!startedInPageEvt) throw new LHError(LHError.errors.NO_TRACING_STARTED);
-    // @ts-ignore - property chain exists for 'TracingStartedInPage' event.
+    // Find out the inspected page frame.
+    let startedInPageEvt;
+    const startedInBrowserEvt = keyEvents.find(e => e.name === 'TracingStartedInBrowser');
+    if (startedInBrowserEvt && startedInBrowserEvt.args.data.persistentIds) {
+      const mainFrame = (startedInBrowserEvt.args.data.frames || []).find(frame => !frame.parent);
+      const mainFrameId = mainFrame && mainFrame.frame || '';
+      startedInPageEvt =
+        {...startedInBrowserEvt, name: 'TracingStartedInPage', args: {data: {page: mainFrameId}}};
+    }
+    // Support legacy browser versions that do not emit TracingStartedInBrowser event.
+    if (!startedInPageEvt) {
+      // The first TracingStartedInPage in the trace is definitely our renderer thread of interest
+      // Beware: the tracingStartedInPage event can appear slightly after a navigationStart
+      startedInPageEvt = keyEvents.find(e => e.name === 'TracingStartedInPage');
+      if (!startedInPageEvt) throw new LHError(LHError.errors.NO_TRACING_STARTED);
+      // @ts-ignore - property chain exists for 'TracingStartedInPage' event.
```

Comment:
two lines above should be pulled out of the conditional (throw if no `startedInPageEvt` found, `// @ts-ignore ` comment is for the `const frameId = ` line below)

Reply:
It doesn't really matter as when we didn't enter the `if` block the `startedInPageEvt` is already there. Anyway, I will move it.